Packs/Flashpoint/Integrations/Ignite/Ignite.py

```diff
@@ -651,13 +651,27 @@ def prepare_incidents_from_compromised_credentials_data(response: dict, next_run
     severity = demisto.params().get('severity', DEFAULT_SEVERITY)
 
     for hit in hits:
-        name = hit.get('_source', {}).get('email')
-        if not name:
-            name = hit.get('_source', {}).get('fpid', 'Compromised Credential Alert')
+        hit_source = hit.get('_source', {})
+        incident_name = ''
+        hit_username = hit_source.get('username')
+        if hit_username:
+            incident_name = hit_username
+            demisto.debug(f'Setting incident name with username: {incident_name}')
+        hit_email = hit_source.get('email')
+        if not incident_name and hit_email:
+            incident_name = hit_email
+            demisto.debug(f'Setting incident name with email: {incident_name}')
+        hit_fpid = hit_source.get('fpid')
+        if not incident_name and hit_fpid:
+            incident_name = hit_fpid
+            demisto.debug(f'Setting incident name with fpid: {incident_name}')
+        if not incident_name:
+            incident_name = 'Compromised Credential Alert'
+            demisto.debug(f'Setting incident name with default: {incident_name}')
         incidents.append({
-            'name': name,
+            'name': incident_name,
             'severity': IncidentSeverity.__dict__.get(severity.upper()),
-            'occurred': hit.get('_source', {}).get('breach', {}).get('created_at', {}).get('date-time'),
+            'occurred': hit_source.get('breach', {}).get('created_at', {}).get('date-time'),
             'rawJSON': json.dumps(hit)
         })
```

Comment:
You can spin this off to use a helper function which might be more performative.

Something like this:
```python
def get_incident_name(hit_source: dict) -> str:
    '''
    Determines the incident name based on available fields in the hit source.
    :param hit_source: The source data from the hit.
    :return: The incident name.
    '''
    for field in ['username', 'email', 'fpid']:
        value = hit_source.get(field)
        if value:
            demisto.debug(f'Setting incident name with {field}: {value}')
            return value
    demisto.debug('Setting incident name with default: Compromised Credential Alert')
    return 'Compromised Credential Alert'


def process_hits(hits: list):
    '''
    Processes the hits and generates incidents with the appropriate severity and details.
    :param hits: The list of hits to process.
    :return: List of incident dictionaries.
    '''
    severity = demisto.params().get('severity', DEFAULT_SEVERITY)
    incidents = []

    for hit in hits:
        hit_source = hit.get('_source', {})
        incident_name = get_incident_name(hit_source)

        incidents.append({
            'name': incident_name,
            'severity': getattr(IncidentSeverity, severity.upper(), None),  # safer access to enum
            'occurred': hit_source.get('breach', {}).get('created_at', {}).get('date-time'),
            'rawJSON': json.dumps(hit)
        })

    return incidents
```

Reply:
Hi @amshamah419, thanks for suggesting helper function. We have pushed a commit for this.